sdks/java/core/src/main/java/org/apache/beam/sdk/testing/CoderProperties.java

```diff
@@ -322,6 +323,26 @@
     }
   }
 
+  public static <T> void coderConsumesBytesEqualToBytesEncoded(Coder<T> coder, T value)
+      throws IOException {
+
+    byte[] bytes = encode(coder, Coder.Context.NESTED, value);
+
+    @SuppressWarnings("unchecked")
+    Coder<T> deserializedCoder = Serializer.deserialize(coder.asCloudObject(), Coder.class);
```

Comment:
We don't need to serialize/deserialize the coder if this gets folded into the decode(Coder, Context, bytes[]) method